purchases/src/test/java/com/revenuecat/purchases/galaxy/utils/ErrorVoUtilsTest.kt

```diff
@@ -34,8 +35,63 @@ class ErrorVoUtilsTest {
         assertThat(errorVo.isError()).isTrue
     }
 
-    private fun mockErrorVo(errorCode: Int): ErrorVo =
+    @Test
+    fun `toPurchasesError returns null when error code indicates no error`() {
+        val errorVo = mockErrorVo(GalaxyErrorCode.IAP_ERROR_NONE.code, "ok")
+
+        val purchasesError = errorVo.toPurchasesError()
+
+        assertThat(purchasesError).isNull()
+    }
+
+    @Test
+    fun `toPurchasesError maps known galaxy error codes to purchases error codes`() {
+        val expectedMappings = listOf(
+            GalaxyErrorCode.IAP_PAYMENT_IS_CANCELED to PurchasesErrorCode.PurchaseCancelledError,
+            GalaxyErrorCode.IAP_ERROR_ALREADY_PURCHASED to PurchasesErrorCode.ProductAlreadyPurchasedError,
+            GalaxyErrorCode.IAP_ERROR_PRODUCT_DOES_NOT_EXIST to PurchasesErrorCode.ProductNotAvailableForPurchaseError,
+            GalaxyErrorCode.IAP_ERROR_ITEM_GROUP_DOES_NOT_EXIST to PurchasesErrorCode.ProductNotAvailableForPurchaseError,
+            GalaxyErrorCode.IAP_ERROR_NOT_EXIST_LOCAL_PRICE to PurchasesErrorCode.ProductNotAvailableForPurchaseError,
+            GalaxyErrorCode.IAP_ERROR_NETWORK_NOT_AVAILABLE to PurchasesErrorCode.NetworkError,
+            GalaxyErrorCode.IAP_ERROR_IOEXCEPTION_ERROR to PurchasesErrorCode.NetworkError,
+            GalaxyErrorCode.IAP_ERROR_SOCKET_TIMEOUT to PurchasesErrorCode.NetworkError,
+            GalaxyErrorCode.IAP_ERROR_CONNECT_TIMEOUT to PurchasesErrorCode.NetworkError,
+            GalaxyErrorCode.IAP_ERROR_WHILE_RUNNING to PurchasesErrorCode.PurchaseInvalidError,
+            GalaxyErrorCode.IAP_ERROR_NEED_APP_UPGRADE to PurchasesErrorCode.StoreProblemError,
+            GalaxyErrorCode.IAP_ERROR_INITIALIZATION to PurchasesErrorCode.StoreProblemError,
+            GalaxyErrorCode.IAP_ERROR_COMMON to PurchasesErrorCode.StoreProblemError,
+            GalaxyErrorCode.IAP_ERROR_CONFIRM_INBOX to PurchasesErrorCode.StoreProblemError,
+            GalaxyErrorCode.IAP_ERROR_NOT_AVAILABLE_SHOP to PurchasesErrorCode.PurchaseNotAllowedError,
+            GalaxyErrorCode.IAP_ERROR_INVALID_ACCESS_TOKEN to PurchasesErrorCode.InvalidCredentialsError,
+        )
+
+        expectedMappings.forEach { (galaxyCode, purchasesErrorCode) ->
```

Comment:
Instead of iterating over the hardcoding mappings here, should we iterate over all values of the `GalaxyErrorcode` enum, so we make sure we catch all of them? Then, we can convert `expectedMappings` to a map, for easier access to the expected PurchasesErrorCode.